pkg/sql/catalog/catalogkeys/keys.go

```diff
@@ -228,3 +154,201 @@ func PrettySpans(index *descpb.IndexDescriptor, spans []roachpb.Span, skip int)
 	}
 	return b.String()
 }
+
+// DatabaseKey implements DescriptorKey.
+type DatabaseKey struct {
+	name string
+}
+
+// NewDatabaseKey returns a new DatabaseKey.
+func NewDatabaseKey(name string) DatabaseKey {
+	return DatabaseKey{name: name}
+}
+
+// Key implements DescriptorKey interface.
+func (dk DatabaseKey) Key(codec keys.SQLCodec) roachpb.Key {
+	return MakeNameMetadataKey(codec, keys.RootNamespaceID, keys.RootNamespaceID, dk.name)
+}
+
+// Name implements DescriptorKey interface.
+func (dk DatabaseKey) Name() string {
+	return dk.name
+}
+
+// TableKey implements DescriptorKey interface.
+type TableKey struct {
+	parentID       descpb.ID
+	parentSchemaID descpb.ID
+	name           string
+}
+
+// NewPublicTableKey returns a new TableKey scoped under the public schema.
+func NewPublicTableKey(parentID descpb.ID, name string) TableKey {
+	return TableKey{parentID: parentID, parentSchemaID: keys.PublicSchemaID, name: name}
+}
+
+// NewTableKey returns a new TableKey
```

Comment:
nit: these are missing some periods

Reply:
added a commit to the end so I can just bors this one.

Reply:
Done